syntax = "proto3";

package Userportal;

import "Versioning.proto";

message Project_portal
{
  optional string project_name = 1;
  optional int32 project_id = 2;
  optional int32 userID = 3;
  repeated File_version versions = 4;
}

message User_data
{
  optional string email = 1;
  optional string password = 2;
  optional string user_name = 3;
  optional int32 user_id = 4;
  optional string location = 5;
  optional int32 age = 6;
  repeated Project_portal projects = 7;
}

message User_file
{
  repeated User_data candidate = 1;
}
